src/lib/intl.d.ts

```diff
@@ -103,7 +108,7 @@ declare module Intl {
     }
 
     interface DateTimeFormat {
-        format(date: number): string;
+        format(date?: Date | number): string;

```

Comment:
Thanks for the review.

My understanding from the described algorithm is the following:
```
Intl.DateTimeFormat.prototype.format(date) {
  if (date === undefined)  // do something
  else x = toNumber(date);
  return FormatDateTime(this, x);
}
```
where [toNumber](http://www.ecma-international.org/ecma-262/5.1/#sec-9.3) tries to convert essentially any object to a number and fails with a RangeError if it can't. In other words the correct type for `date` would be `any` according to the specs and the output of the following are the same in all of the browsers that implemented this feature:
`new Intl.DateTimeFormat().format(123456789)`
`new Intl.DateTimeFormat().format('123456789')`

I'm not sure how would you normally treat these cases in d.ts files because I'm sure many functions actually do these toNumber, toBoolean, ... on their inputs as javascript doesn't have a type system by nature. My feeling was you will try to enforce type as much as it makes sense to ensure code sanity, etc. If you think `any` would be the way to go here, let me know.